### DISCUSSION TITLE
Do you know of a way to not start USB hub at boot up?

### DISCUSSION THREAD

**Discussion by cainbantam** [mvp/uhubctl#314] [Q&A] [answered]
uhubctl is blocked by an USB DVD drive which drains too much power and needs an additionnal power source. Since the additional power source isn't usually turned on, the DVD drive fails at starting but keeps trying and prevents uhubctl from stopping the USB hub. If I could just prevent the USB hub from starting at boot, then I could prevent my DVD drive from starting altogether, and then set it on on will with usbhub and turning on its additional power source... Any advice welcome. Otherwise, I'll just unplug the DVD drive :)

**Comment by mvp** [comment, marked as answer]
You can add something like this to /etc/rc.local:

    uhubctl -l 1-1 -p 2 -a 0

This will turn power off as soon as possible at boot up. However, Linux kernel will enable power even before processing /etc. Just about the only way to achieve that would be patching Linux kernel USB driver to not enable power at boot.

**Reply by cainbantam** [reply]
thanks, it is how I used to turn off the usb hub at startup originally. Yet it isn't working with that DVD player plugged in. I thought first it was maybe because I then activated the desktop mode (instead of the usual console mode I use when my RPi works as a server and not as a DVD player...) and I tried to replace the line in rc.local with a systemd unit since I have heard that rc.local had compatibility issues with raspbian buster, but so far to no avail. When I let the additional power source for the USB DVD player on, uhubctl works properly, but the point of stopping the usb hub is to save energy, so it is not a relevant solution. I will try different configurations and give you an update...

**Reply by cainbantam** [reply]
One question concerning your answer: why the "-p 2" in your command line above? I thought you recommended to write "-p 1-4" usually, for the USB ports could not be turned off individually on RPi4?

**Reply by mvp** [reply]
Exact command depends on RPI model. FAQ suggests commands for each model, they are slightly different. Note that -p 1-4 is the same as -p all, or not specifying -p option at all.

**Reply by SlySven** [reply]
IIRC On early RPis 1B the Ethernet port is connected to p1 and it is power-switched separately to the power-ganged USB ports on p2 and p3 (controlled via p2 power control).  You do not want to switch off p1 if you are SSHing into your RPi via a wired LAN...!

**Reply by ghost** [reply]
I'm trying to come up with a persistent way to disable it. the Bigtreetech SKR Mini e3 boards (v2 and prior - current as of this writing) will be backpowered by the port USB and this can actually create a dangerous situation.
